packages/pnpm/src/main.ts

```diff
@@ -134,6 +134,10 @@ export default async function run (inputArgv: string[]) {
     }
   }
 
+  if (!config.recursiveInstall) {
```

Comment:
I think we should move this code to main.ts:

https://github.com/pnpm/pnpm/blob/e6e50013058e8f44d378d0a84a7b23ace042fad6/packages/parse-cli-args/src/index.ts#L121-L127

In that case, we can add an additional check for this setting to that if